.github/workflows/pre-merge.yml

```diff
@@ -1,6 +1,6 @@
 name: Pre-Merge Checks
 
-on: [push, pull_request]
+on: [pull_request]
```

Comment:
No more ninja push? 😅

Reply:
So when you push commits to an open merge request the action will still run. That's what I was guessing you meant by `ninja push`.

This just removes the action running on any pull request. That way when I push commits to a remote feature branch (that hasn't had a merge request open). I don't get emails for failed builds, because I am just working on a branch not ready for review yet.

Reply:
> That's what I was guessing you meant by ninja push.

Whoops, it's my internal brain process that's not documented 😅. So what I mean by `ninja push`, we currently create pull requests for notable changes, and directly push to version branch (`v1.3`) when for example there's a typo, thus avoiding the `Pre-Merge Checks` if we do it, so do you suggest if we avoid doing it? and for every change we need to apply we have to create a PR for that.

Reply:
I think it makes sense to apply the `push` rules to branches where it makes sense, like the default branch and version branches. However, I think it shouldn't apply to issue branches.

Reply:
A rule can be added to match on branch pushes, if that sounds good.

Reply:
>  like the default branch and version branches. However, I think it shouldn't apply to issue branches.

*dumb question*: does this event only triggered once the PR is created? as we're still pushing things even after the PR is created.

Reply:
If the `pull_request` option is only set, then the action will always run on the head commit of the branch that has a PR submitted. So if commits are push after the PR is opened, the action will run for the latest head commit. Was that what you where wondering?

Reply:
Yes! It's now clear to me